.github/workflows/Cargo.yml

```
@@ -0,0 +1,45 @@
+name: Cargo CI
+
+on: [push, pull_request]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  fmt:
+    name: Rustfmt Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: stable
+          components: rustfmt
+      - name: Run cargo fmt
+        run: cargo fmt --all -- --check
+
+  test:
+    name: Run Tests
+    needs: fmt
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        PACKAGE: [artinchip-rt, artinchip-hal]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: stable
+      - name: Run cargo test
+        run: cargo test -p ${{ matrix.PACKAGE }}
+  clippy:
+    name: Clippy Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: stable
+          components: clippy
+      - name: Run cargo clippy
+        run: cargo clippy --all-targets --all-features -- -D warnings
```

```suggestion
        run: cargo clippy --all-targets --all-features -- -D warnings

```
Add a new line at the end of each file.